### ISSUE TITLE
Isn't Date a Binding?

### ISSUE THREAD

**Issue by OrkhanAlikhanov** [stephencelis/SQLite.swift#658]
I think it is otherwise it would not compile. However, it is interpreted as NULL.

``` swift
extension ExpressionType where UnderlyingType == Date {
    func like(_ date: Date) -> Expression<Bool> {
        let castedSelf = self as! Expression<Date>
        let a = date.startOfDay...date.endOfDay ~= castedSelf
        print(a.asSQL()) //prints "date" BETWEEN NULL AND NULL

        let b = castedSelf >= date.startOfDay && castedSelf <= date.endOfDay
        print(b.asSQL()) // prints correctly (("date" >= '2017-05-26T00:00:00.000') AND ("date" <= '2017-05-26T23:59:59.000'))
        return b 
    }
}
```

**Comment by jberkel** [comment]
Fixed in 364a240a6217d4d1b6fd7c237a3cf9039cbad6ec

**Comment by jberkel** [comment]
Part of 0.11.4
